storage_controller/src/service.rs

```diff
@@ -4249,7 +4255,30 @@ impl Service {
         };
 
         let waiter_count = waiters.len();
-        self.await_waiters(waiters, RECONCILE_TIMEOUT).await?;
+        match self.await_waiters(waiters, RECONCILE_TIMEOUT).await {
+            Ok(()) => {}
+            Err(ReconcileWaitError::Failed(_, reconcile_error))
+                if matches!(*reconcile_error, ReconcileError::Cancel) =>
+            {
+                // Ignore reconciler cancel errors: this reconciler might have shut down
+                // because some other change superceded it.  We will return a nonzero number,
+                // so the caller knows they might have to call again to quiesce the system.
+            }
+            Err(ReconcileWaitError::Failed(_, reconcile_error)) => {
+                tracing::error!("Reconcile error in reconcile_all_now: Failed but not Cancel: {reconcile_error}");
```

Comment:
Shouldn't this `return Err(...)`? I think that's why the test is failing.

Reply:
You're right.  This was a piece of debug that I committed accidentally.

d5dc5cbdf